### PR TITLE
[SPARK-37951][MLLIB][K8S] Move test file from ../data/ to corresponding module's resource folder

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Move test file form `data/` dir to corresponding module's resource folder

1. move  `../data/mllib/images/partitioned` to mllib's `resources/inages/partitioned`
2. move `../data/mllib/iris_libsvm.txt`  to mllib's `resources/iris_libsvm.txt`
3. copy `data/mllib/pagerank_data.txt` to  kubenets-integration-test's `resources/pagerank_data.txt`

### Why are the changes needed?
Refactor code to avoid test failure 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existed UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan

**Comment by cloud-fan** [comment]
@AngersZhuuuu are you sure these files are only used in tests?

also cc @WeichenXu123 , I saw this test suite failed a few times due to file not found. I think it's better to access test files under the resource directory.

**Comment by AngersZhuuuu** [comment]
> @AngersZhuuuu are you sure these files are only used in tests?

Change the doc

**Comment by srowen** [comment]
I agree in any event and if so I'd love to see the rest of the files there moved, indeed.

**Comment by cloud-fan** [comment]
@AngersZhuuuu are these the only testing data files? I think we should only put data files for `examples` in that directory.

**Comment by HyukjinKwon** [comment]
I think some of them are also used in examples and PySpark test cases IIRC. We will have to better identify ones not used in examples, and move them to resource directories for now.

**Comment by AngersZhuuuu** [comment]
How about current

**Comment by srowen** [comment]
Looks good if that's all the files and it works!

File: python/pyspark/ml/tests/test_image.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 class ImageFileFormatTest(SparkSessionTestCase):
     def test_read_images(self):
-        data_path = "data/mllib/images/origin/kittens"
+        data_path = "python/test_support/images/kittens"
```
Comment (cloud-fan) [inline python/pyspark/ml/tests/test_image.py:27]: does pyspark have the test resource directory? If not I think it's fine to reuse the data files in `/data/mllib/images/...`

Comment (HyukjinKwon) [reply python/pyspark/ml/tests/test_image.py:27]: yeah, pyspark doesn't have test resource directory.

Comment (AngersZhuuuu) [reply python/pyspark/ml/tests/test_image.py]: Revert this change.

**Review by cloud-fan — APPROVED** [review]

**Comment by AngersZhuuuu** [comment]
> Looks good if that's all the files and it works!

All test passed!

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**